influxql/query_executor.go

```diff
@@ -349,44 +349,50 @@ func (e *QueryExecutor) query(qid uint64) (*QueryTask, bool) {
 //
 // After a query finishes running, the system is free to reuse a query id.
 func (e *QueryExecutor) attachQuery(q *Query, database string, interrupt <-chan struct{}) (uint64, *QueryTask, error) {
-	e.mu.Lock()
-	defer e.mu.Unlock()
+	qid, task, err := func() (uint64, *QueryTask, error) {
```

Comment:
OOM is more likely to kill the whole process, and not something you're going to be able to recover from anyway. The allocation of the `QueryTask` should probably be done before the lock, anyway, to keep the time the lock is held to a minimum. Which would make the lock and all possible unlocks span about 10 lines. We've generally done this with lock contentions and allocations in other parts of the code, I don't see why this case should be special.